### PR TITLE
Product

### PR DESCRIPTION
**Pull request by NikitaBondarevv**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  render() {
+    return (
+      <>
+        <Header />
+        <Main />
+      </>
+    )
+  }
+}
```
Comment (iliatsymbal) [inline index.js]: what the reason to use class here?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "todo",
```
Comment (iliatsymbal) [inline package.json]: use an appropriate name for the project

File: package.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "webpack serve --open",
+    "lint": "eslint src/**/**.js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/NikitaBondarevv/todo.git"
```
Comment (iliatsymbal) [inline package.json]: wrong url

File: package.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/NikitaBondarevv/todo/issues"
+  },
+  "homepage": "https://github.com/NikitaBondarevv/todo#readme",
```
Comment (iliatsymbal) [inline package.json]: wrong urls

File: src/header/styles.css [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+header .list {
+  list-style: none;
+  margin: 0;
+  padding: 0;
+  display: flex;
+  column-gap: 68px;
+}
+
```
Comment (iliatsymbal) [inline src/header/styles.css]: `header` you don't need it here

File: src/main/index.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { NumberOfProductsAndCategories } from '../numberOfProductsAndCategories'
+import { products } from '../helpers/products'
+import styles from './styles.css'
+import { ProductInformation } from '../productInformation'
+
+export const Main = () => (
+  <main className={styles.main}>
+    <NumberOfProductsAndCategories name='Nikita' products={products} />
```
Comment (iliatsymbal) [inline src/main/index.js]: it's something like `ProductsInfo` or `GeneralInfo` or just `Info`

File: src/main/index.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    <ProductInformation />
+  </main>
+)
```
Comment (iliatsymbal) [inline src/main/index.js]: empty line

File: src/main/styles.css [hunk 8]
```diff
@@ -0,0 +1,4 @@
+main {
+  display: flex;
+  justify-content: space-around;
+}
```
Comment (iliatsymbal) [inline src/main/styles.css]: empty line

File: src/numberOfProductsAndCategories/index.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
+import styles from './styles.css'
+
+export const NumberOfProductsAndCategories = ({ name, products }) => {
+  const productsAndCategories = products
```
Comment (iliatsymbal) [inline src/numberOfProductsAndCategories/index.js]: what the reason of this expression?

File: src/numberOfProductsAndCategories/index.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  
+  const getNumberOfProducts = () => {
+    return productsAndCategories.map(product => product.product)
+  }
+  
+  const getNumberOfPublished = () => {
+    const published = []
+
+    productsAndCategories.map(product => {
+      if (product.published === true) {
```
Comment (iliatsymbal) [inline src/numberOfProductsAndCategories/index.js]: `if (product.published) {`

File: src/numberOfProductsAndCategories/index.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  }
+  
+  const getNumberOfProducts = () => {
+    return productsAndCategories.map(product => product.product)
+  }
+  
+  const getNumberOfPublished = () => {
+    const published = []
+
+    productsAndCategories.map(product => {
```
Comment (iliatsymbal) [inline src/numberOfProductsAndCategories/index.js]: why do you use `map` here?

File: src/numberOfProductsAndCategories/index.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    <div className={styles.productsInfo} >
+      <span className={styles.greetings} >Hello, {name}</span>
+      <ul className={styles.list}>
+        <li>
+          You have {getNumberOfCategories().length} categories ({getNumberOfPublished().length} published)
+        </li>
+        <li>
+          You have {getNumberOfProducts().length} products
+        </li>
+      </ul>
```
Comment (iliatsymbal) [inline src/numberOfProductsAndCategories/index.js]: 1) I don't see why it is a `ul`
2) numbers should be bold

File: src/productInformation/index.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import { Component } from 'react'
+import { EditableText } from './editableText'
+
+import styles from './styles.css'
+
+const descriptionText = 'Enjoy new movie-themed thrills ad a favorite Radiator Springs location - Willys Butte! Inspired by Disney Pixar Cars 3, this hair-raising set uses multiple wild driving skills and transforms for 3 ways play.'
+
+export class ProductInformation extends Component {
+
+  render() {
```
Comment (iliatsymbal) [inline src/productInformation/index.js]: no need class here

File: src/productInformation/index.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+
+    return (
+      <div className={styles.productInformation}>
+        <span className={styles.title}>TITLE:<EditableText text="TOW TRACK" stylesInput={styles.editableTitle} /></span>
+        <span className={styles.price}>$<EditableText text="32" stylesInput={styles.editablePrice} /></span>
```
Comment (iliatsymbal) [inline src/productInformation/index.js]: there should be a space between `$` and price

File: index.js (hunk 1 above)
Comment (NikitaBondarevv) [reply index.js]: This is a useless use. Corrected.

File: package.json (hunk 2 above)
Comment (NikitaBondarevv) [reply package.json]: Oh sorry. Okay i see it.

File: package.json (hunk 3 above)
Comment (NikitaBondarevv) [reply package.json]: Yess. Fixed this

File: package.json (hunk 4 above)
Comment (NikitaBondarevv) [reply package.json]: Fixed

File: src/header/styles.css (hunk 5 above)
Comment (NikitaBondarevv) [reply src/header/styles.css]: Okay. Thank you. Fixed it

File: src/main/index.js (hunk 6 above)
Comment (NikitaBondarevv) [reply src/main/index.js]: Thank you for name. I change it

File: src/productInformation/editableText/index.js [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+  const showInput = () => {
+    setHidden(false)
+  }
+
+  return (
+    hidden
+      ? <span className={styles.text} onClick={showInput}>{value ? value : text}</span>
+      : (
+        <form className={styles.editableText} onSubmit={handleBlur}>
```
Comment (iliatcymbal) [inline src/productInformation/editableText/index.js]: do not use form

File: src/components/header/index.js [hunk 16]
```diff
@@ -0,0 +1,36 @@
+import styles from './styles.css'
+import logo from './images/logo.png'
+import loginPage from './images/loginPage.png'
+
+export const Header = () => {
+  const links = [
+    { text: 'Home', value: 'home' },
+    { text: 'Shop', value: 'shop' },
+    { text: 'Contacts', value: 'contacts' }
+  ]
```
Comment (iliatcymbal) [inline src/components/header/index.js]: I don't see a reason why you're keeping it inside component

File: src/components/header/index.js [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+
+  return (
+    <header className={styles.header}>
+      <div className={styles.navigation}>
```
Comment (iliatcymbal) [inline src/components/header/index.js]: `div` It us an extra tag.  Every time try to use as minimal tags as possible

File: src/components/productInformation/editableText/index.js [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+  const showInput = () => {
+    setHidden(false)
+  }
+
+  return (
+    hidden
+      ? <span className={styles.text} onClick={showInput}>{value ? value : text}</span>
+      : multiLine
+        ? (
+          <textarea
```
Comment (iliatcymbal) [inline src/components/productInformation/editableText/index.js]: do not nest ternary operators

**Pull request merged**